src/app/shared/directives/change-background.directive.ts

```diff
@@ -0,0 +1,29 @@
+import { AfterViewInit, Directive, ElementRef, HostBinding, HostListener, Input } from '@angular/core';
+import { HOVER_BACKGROUND_COLOR } from '../constants/colors';
+
+@Directive({
+  selector: '[appBackGroundOnHover]',
+})
+export class ChangeBackgroundDirective implements AfterViewInit {
+  @Input('appBackGroundOnHover') backgroundColor: string = HOVER_BACKGROUND_COLOR;
+
+  private initialColor!: string;
+
+  constructor(private element: ElementRef<HTMLElement>) {}
+
+  @HostBinding('style.backgroundColor') elementBackground!: string;
+
+  ngAfterViewInit(): void {
+    this.initialColor = this.element.nativeElement.style.backgroundColor;
+  }
+
+  @HostListener('mouseover')
```

Comment:
The pairs are `mouseover + mouseout` or `mouseenter + mouseleave`, they shouldn't be mixed
https://learn.javascript.ru/mousemove-mouseover-mouseout-mouseenter-mouseleave

`mouseenter + mouseleave` seems to be more suitable here

Reply:
Thanks for finding this confusion, sorry. 
Fixed